README.md

```diff
@@ -1,111 +1,116 @@
+an example):
+
+```yaml
+install:
+  - nomad:
+      jobs:
+        - path: nomad/pytechco-redis.nomad.hcl
+        - path: nomad/pytechco-web.nomad.hcl
+        - path: nomad/pytechco-setup.nomad.hcl
+        - dispatch: pytechco-setup
+          meta:
+            budget: "200"
+        - path: nomad/pytechco-employee.nomad.hcl
+```
+
+The job step supports the following 3 actions which can be specified as a list item in the `jobs:` block:
+
+- `path`: runs a nomad job file using the given nomad hcl file.
+- `dispatch`: dispatches a parametrized nomad job using the given job name.
+    - `meta`: (optional): a map of key/value pairs that will be passed as metadata to the nomad job.
+    - `idPrefixTemplate` (optional): a prefix added to dispatched job IDs.
+    - `payload` (optional): base64 encoded string containng the payload to pass to the nomad job. Limited to 65536 bytes.
+- `stop`: stops a nomad job using the given job name.
+    - `purge`: (optional): if set to true, the job will be purged immediately.
+
+Each of the 3 actions above can also take in the following options, which can be passed as arguments to each individual
+nomad run, overriding any global nomad parameters set earlier:
+
+- `address` (string): the address of the nomad server to connect to.
+- `region` (string): the region of the nomad server to connect to.
+- `namespace` (string): the namespace of the nomad server to connect to.
+- `caCert` (string): the path to the CA cert file to use for TLS.
+- `caPath` (string): the path to a directory of CA cert files to use for TLS.
+- `clientCert` (string): the path to the client cert file to use for TLS.
+- `clientKey` (string): the path to the client key file to use for TLS.
+- `tlsServerName` (string): the server name to use as the SNI host when connecting via TLS.
+- `tlsSkipVerify` (bool): disables TLS host verification.
+- `token` (string): the ACL token to use when connecting to Nomad.
+
+Additionally, the mixin also supports porter outputs. To specify an output for a job, add the `outputs` block to your job, with a
+list of `name/key` pairs to your run like so
+
+```yaml
+nomad:
+  jobs:
+    - path: nomad/pytechco-redis.nomad.hcl
+      outputs:
+        - name: redis-evalId 
+          key: evalId
+```
+
+You can also specify a top-level `outputs` block in the porter.yaml, to specify the outputs of the bundle as a whole
+
+```yaml
+outputs:
+  - name: redis-evalId
+    type: string
+  - name: web-evalId
+    type: string
+```
+
+This output can then be used in subsequent steps in your porter.yaml file using the syntax ${  bundle.outputs.redis-evalId }, see 
+the porter [documentation](https://porter.sh/wiring/#wiring-outputs) for more details.
+
+Currently, the only supported output key is `evalId`, which is the ID of the nomad evaluation that was created when the job was run. 
+Note that not all nomad jobs will create an evaluation, so this output will only be available for jobs that do.
```

Comment:
In future iterations, for jobs of `type="service"`, you'll almost certainly want to use that evaluation ID to make a blocking query for a deployment ID to be created, and then monitor the progress of that deployment. Otherwise users are likely to do something like:

```yaml
nomad:
  jobs:
    - path: ./database.nomad.hcl
    - path: ./application.nomad.hcl
```

Where the application assumes that the database is already up and running and fails if it's not up yet. 

Scheduling in Nomad is asynchronous. So registering a job and getting the evaluation ID back doesn't necessarily mean the job will ever result in running allocations! For example, you could get a deployment ID back but it turns out no Nomad clients (nodes where work is deployed) can actually support the task because they don't have the right task driver or whatever. That doesn't get determined until the Evaluation is processed by the scheduler.

Reply:
I have indeed noticed that the api is completely async, and I was wondering if the api at all allows for blocking calls. Do you think adding a -wait flag (opposite of -detach) would be a good idea? I did see that the terraform nomad provider did something similar to what you mentioned https://github.com/hashicorp/terraform-provider-nomad/pull/64

Reply:
Seem like a reasonable approach